lib/src/Base/Func/IndicatorEvaluation.cxx

```diff
@@ -22,35 +22,32 @@
 
 #include "openturns/IndicatorEvaluation.hxx"
 #include "openturns/PersistentObjectFactory.hxx"
-#include "openturns/Less.hxx"
+#include "openturns/Interval.hxx"
 
 BEGIN_NAMESPACE_OPENTURNS
 
+typedef Collection<UnsignedInteger>     UnsignedIntegerCollection;
+
 CLASSNAMEINIT(IndicatorEvaluation)
 
 static const Factory<IndicatorEvaluation> Factory_IndicatorEvaluation;
 
 /* Default constructor */
 IndicatorEvaluation::IndicatorEvaluation()
   : EvaluationImplementation()
-  , evaluation_()
-  , comparisonOperator_()
-  , threshold_(0.0)
+  , domain_(Interval())
 {
   // Nothing to do
 } // IndicatorEvaluation
 
 /* Default constructor */
-IndicatorEvaluation::IndicatorEvaluation(const Evaluation & evaluation,
-    const ComparisonOperator & comparisonOperator,
-    const Scalar threshold)
+IndicatorEvaluation::IndicatorEvaluation(const Domain & domain)
   : EvaluationImplementation()
-  , evaluation_()
-  , comparisonOperator_(comparisonOperator)
-  , threshold_(threshold)
+  , domain_(domain)
 {
-  setEvaluation(evaluation);
-  setDescription(evaluation.getDescription());
+  const UnsignedInteger domainDimension = domain_.getDimension();
+  const Sample dummySample(1, domainDimension);
+  EvaluationImplementation::setInputDescription(dummySample.getDescription());
```

Comment:
Remove `dummySample` and use `Description::BuildDefault(domainDimension, "v")` instead to build the description.